spring-rabbit/src/test/java/org/springframework/amqp/rabbit/annotation/EnableRabbitIdleContainerTests.java

```diff
@@ -0,0 +1,136 @@
+import org.springframework.amqp.rabbit.core.RabbitAdmin;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.amqp.rabbit.listener.ListenerContainerIdleEvent;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+/**
+ * @author Gary Russell
+ * @since 1.6
+ *
+ */
+@ContextConfiguration
+@RunWith(SpringJUnit4ClassRunner.class)
+@DirtiesContext
+public class EnableRabbitIdleContainerTests {
+
+	@Autowired
+	private Listener listener;
+
+	@Autowired
+	private RabbitTemplate rabbitTemplate;
+
+	@Autowired
+	private Queue queue;
+
+	@Test
+	public void testIdle() throws Exception {
+		assertEquals("FOO", this.rabbitTemplate.convertSendAndReceive(this.queue.getName(), "foo"));
+		assertTrue(this.listener.latch.await(10, TimeUnit.SECONDS));
+		assertEquals("foo", this.listener.event.getListenerId());
+		assertEquals(this.queue.getName(), this.listener.event.getQueues()[0]);
+		assertEquals("BAR", this.rabbitTemplate.convertSendAndReceive(this.queue.getName(), "bar"));
+	}
+
+	@Configuration
+	@EnableRabbit
+	public static class EnableRabbitConfig {
+
+		@Bean
+		public SimpleRabbitListenerContainerFactory rabbitListenerContainerFactory() {
+			SimpleRabbitListenerContainerFactory factory = new SimpleRabbitListenerContainerFactory();
+			factory.setConnectionFactory(rabbitConnectionFactory());
+			factory.setNoMessageAlertInterval(500L);
+			factory.setReceiveTimeout(100L);
+			return factory;
+		}
+
+		@Bean
+		public ConnectionFactory rabbitConnectionFactory() {
+			CachingConnectionFactory connectionFactory = new CachingConnectionFactory();
+			connectionFactory.setHost("localhost");
+			return connectionFactory;
+		}
+
+		@Bean
+		public RabbitTemplate rabbitTemplate() {
+			return new RabbitTemplate(rabbitConnectionFactory());
+		}
+
+		@Bean
+		public RabbitAdmin rabbitAdmin() {
+			return new RabbitAdmin(rabbitConnectionFactory());
+		}
+
+		@Bean
+		public Queue queue() {
+			return new AnonymousQueue();
+		}
+
+		@Bean
+		public Listener listener() {
+			return new Listener();
+		}
+
+	}
+
+	public static class Listener implements ApplicationListener<ListenerContainerIdleEvent> {
+
+		private final CountDownLatch latch = new CountDownLatch(2);
+
+		private volatile ListenerContainerIdleEvent event;
+
+		@RabbitListener(id="foo", queues="#{queue.name}")
+		public String listen(String foo) {
+			return foo.toUpperCase();
+		}
+
+		@Override
+		public void onApplicationEvent(ListenerContainerIdleEvent event) {
+			LogFactory.getLog(this.getClass()).info(event);
+			if ("foo".equals(event.getListenerId())) {
```

Comment:
I added it since I am planning on using this as an example in the documentation - where we need to make it clear that events for all containers will be received.